content/en/debugging.md

```diff
@@ -0,0 +1,14 @@
+---
+title: Debugging
+status: Feedback Appreciated
+category: concept
+---
+
+## What it is
+Debugging is the process or activity of finding and resolving bugs (or errors) from computer programs, software, or systems to get the desired output. A bug is a defect or a problem leading to incorrect or unexpected output.
+
+## Problem it addresses
+Developers write source code and create a binary file with external libraries to run it in production. If a bug is introduced during that process, the code will likely not function as desired (undefined behavior) when executed. To identify the source of the problem, developers have to review the code line-by-line. Catching errors is quite hard without context, i.e., variables, data, location in memory, etc.
```

Comment:
Just realized one more thing, sorry. Here we determined that the problem is that we don't have any context to debug an app manually.

Presumably, debugging tools solve that by providing that needed context but we don't mention it below...or at least it's not clear how that relates to things like variables, data, location in memory, etc. 

How it helps should directly address the problem. It's probably already there but implied. We should spell it out.